### PR TITLE
Implement celldesginer file import and export features under File menu

### LINKED ISSUE #242: CellDesigner export/import
Let's use Ludovic's library and a new service for this library to add export/import facilities for CellDesigner maps to Newt.

### PR DESCRIPTION
**Pull request by gokberkkaraca**
Resolves #242

### CODE REVIEW

File: app/js/backbone-views.js [hunk 1]
```diff
@@ -1498,6 +1499,13 @@ var FileSaveView = Backbone.View.extend({
         delete properties.mapType; // already stored in sbgn file, no need to store in extension as property
         chiseInstance.saveAsSbgnml(filename, version, renderInfo, properties);
       }
+      else if(fileformat === "celldesigner") {
+        var celldesignerText = text;
+        var blob = new Blob([celldesignerText], {
```
Comment (kinimesi) [inline app/js/backbone-views.js]: @gokberkkaraca do you really need `celldesignerText`? Provide `text` directly to `Blob`.

File: app/js/app-menu.js [hunk 2]
```diff
@@ -32,6 +32,32 @@ module.exports = function() {
...
+  function cd2sbgnml(xml) {
+
+    $.ajax({
+        type: 'post',
+        url: "http://localhost:4567/cd2sbgnml",
+        data: {xml: xml},
+        success: function (data) {
+            var chiseInstance = appUtilities.getActiveChiseInstance();
+            validateSBGNML(xml);
+            chiseInstance.loadSBGNMLText(data);
```
Comment (kinimesi) [inline app/js/app-menu.js]: @gokberkkaraca 
- spinner should be started before loading and stopped after loading
-  `sbgnvizLoadFileStart` and `sbgnvizLoadFileEnd` events should be triggered when loading starts and finishes, [see this](https://github.com/iVis-at-Bilkent/sbgnviz.js/blob/205fad5a9853d31d2372b9cb2525662b8251bcf0/src/utilities/file-utilities-factory.js#L182)

File: app/js/app-menu.js [hunk 3]
```diff
@@ -32,6 +32,32 @@ module.exports = function() {
       }
     });
   }
+
+  function cd2sbgnml(xml) {
+
+    $.ajax({
+        type: 'post',
+        url: "http://localhost:4567/cd2sbgnml",
```
Comment (kinimesi) [inline app/js/app-menu.js]: should not be hard-coded.

File: app/js/app-menu.js [hunk 4]
```diff
@@ -32,6 +32,32 @@ module.exports = function() {
...
+            chiseInstance.loadSBGNMLText(data);
+        }
+    })
+  }
+
+  function sbgnml2cd(xml) {
+
+      $.ajax({
+          type: 'post',
+          url: "http://localhost:4567/sbgnml2cd",
```
Comment (kinimesi) [inline app/js/app-menu.js]: should not be hard-coded.

File: app/js/app-menu.js [hunk 5]
```diff
@@ -258,9 +284,27 @@ module.exports = function() {
...
 
+    $('#import-celldesigner-file').click(function (){
+        $("#celldesigner-file-input").trigger('click');
+    });
+
+    $('#celldesigner-file-input').change(function (e, fileObject) {
+
+      if ($(this).val() != "" || fileObject) {
+        var file = this.files[0] || fileObject;
+        document.getElementById('file-name').innerHTML = file.name;
```
Comment (kinimesi) [inline app/js/app-menu.js]: @gokberkkaraca no need for this, when the `sbgnvizLoadFileEnd` is triggered properly.

**Comment by gokberkkaraca** [comment]
@kinimesi I fixed the problems you mentioned, can you review again please?

**Pull request merged**